src/managementEnforcer.ts

```diff
@@ -273,6 +278,11 @@ export class ManagementEnforcer extends InternalEnforcer {
    * @return succeeds or not.
    */
   public async addNamedPolicies(ptype: string, rules: string[][]): Promise<boolean> {
+    if (this.autoNotifyWatcher) {
```

Comment:
Sometimes watcher is checked, sometimes not. Reason?

Reply:
If watcher Ex exists, then I think it is better to call watcherEx and not watcher, im checking it afterwards